test/acceptance.sh

```diff
@@ -184,3 +186,34 @@ for scenario in $(echo ${scenarios}); do
 
   popd > /dev/null
 done
+
+# finally do some shellcheck linting
+if [[ "$(command -v shellcheck)" ]]; then
+  echo "==> Shellcheck"
+  pushd "${__root}" > /dev/null
+
+  failed="false"
+
+  while IFS=$'\n' read -r file; do
+    [[ "$file" != "./main.sh" ]] && continue #XXX
```

Comment:
just saw your top level comment.... yeah a bunch of stuff will need to be cleaned up... Also, I'm guessing shellcheck on OS X will be a no go... last time I tried installing with homebrew I ended up having to build from source...